src/Microsoft.Identity.Web.MicrosoftGraph/BaseRequestExtensions.cs

```diff
@@ -15,29 +16,34 @@ public static class BaseRequestExtensions
         /// </summary>
         /// <param name="baseRequest">The <see cref="IBaseRequest"/>.</param>
         /// <param name="scopes">Microsoft graph scopes used to authenticate this request.</param>
-        public static T WithScopes<T>(this T baseRequest, string[] scopes) where T : IBaseRequest
+        public static T WithScopes<T>(this T baseRequest, params string[] scopes) where T : IBaseRequest
         {
-            string authHandlerOptionKey = typeof(TokenAcquisitionAuthenticationProviderOption).ToString();
-            AuthenticationHandlerOption authHandlerOptions = baseRequest.MiddlewareOptions[authHandlerOptionKey] as AuthenticationHandlerOption ?? new AuthenticationHandlerOption();
-            TokenAcquisitionAuthenticationProviderOption msalAuthProviderOption = authHandlerOptions?.AuthenticationProviderOption as TokenAcquisitionAuthenticationProviderOption ?? new TokenAcquisitionAuthenticationProviderOption();
-
-            msalAuthProviderOption.Scopes = scopes;
-
-            authHandlerOptions.AuthenticationProviderOption = msalAuthProviderOption;
-            baseRequest.MiddlewareOptions[authHandlerOptionKey] = authHandlerOptions;
-
-            return baseRequest;
+            return SetParameter(baseRequest, options => options.Scopes = scopes);
         }
 
+        /// <summary>
+        /// Applied to a request, expresses to use App only permissions for Graph
+        /// </summary>
+        /// <typeparam name="T">Type of the request</typeparam>
+        /// <param name="baseRequest">Request</param>
+        /// <param name="appOnly">Should the permissions be app only or not</param>
+        /// <returns></returns>
         public static T WithAppOnly<T>(this T baseRequest, bool appOnly = true) where T : IBaseRequest
         {
-            string authHandlerOptionKey = typeof(TokenAcquisitionAuthenticationProviderOption).ToString();
+            return SetParameter(baseRequest, options => options.AppOnly = appOnly);
+        }
+
+        private static T SetParameter<T>(T baseRequest, Action<TokenAcquisitionAuthenticationProviderOption> action) where T : IBaseRequest
+        {
+            string authHandlerOptionKey = typeof(AuthenticationHandlerOption).ToString();
             AuthenticationHandlerOption authHandlerOptions = baseRequest.MiddlewareOptions[authHandlerOptionKey] as AuthenticationHandlerOption ?? new AuthenticationHandlerOption();
             TokenAcquisitionAuthenticationProviderOption msalAuthProviderOption = authHandlerOptions?.AuthenticationProviderOption as TokenAcquisitionAuthenticationProviderOption ?? new TokenAcquisitionAuthenticationProviderOption();
 
-            msalAuthProviderOption.AppOnly = appOnly;
+            action(msalAuthProviderOption);
 
+#pragma warning disable CS8602 // Dereference of a possibly null reference. The Graph SDK ensures it exists
             authHandlerOptions.AuthenticationProviderOption = msalAuthProviderOption;
```

Comment:
The Graph SDK does add the authHandlerOptions in the properties of the GraphContext